ChuckNorrisFacts/Domain/Service/FactService.swift

```diff
@@ -6,14 +6,27 @@
 //  Copyright © 2018 Wanis. All rights reserved.
 //
 
+import Alamofire
 import Foundation
 import RxSwift
-import Alamofire
 
-struct FactService {
-    func get() {
-        Alamofire.request("https://api.chucknorris.io/jokes/random").responseData { response in
-            
-        }
+protocol FactServiceProtocol {
+    func getFacts(term: String) -> Observable<Facts>
+}
+
+struct FactService: FactServiceProtocol {
+    let map: FactsMapperProtocol
+    
+    func getFacts(term: String) -> Observable<Facts> {
+        return Alamofire.request("https://api.chucknorris.io/jokes/search?query=\(term)").rx.responseData()
+            .retry(1)
```

Comment:
- Qual a diferença entre `retry`e `replay` ?
- Como esse timeout de 60 segundos se relaciona com o valor de timeout default utilizado pela layer de transporte abstraída pelo Alamofire?
- Pq você está executando o timeout na thread principal da aplicação ao invés de uma thread em bg?

Reply:
1. O `retry` acontece quando o evento for `onError`,  ele vai tentar rodar mais `x` vez(es) a subscription na tentativa de que a próxima seja bem sucedida.
O `replay` garante que todos os `Observables` recebam vejam a mesma sequencia de itens emitidos mesmo se algum dos `Observables` já tiver emitido algum item.

2. Ambos são de 60 segundos, acho que fica desnecessário deixar esse timeout aí. Vou tirar.

3. Não sei dizer, é melhor usar em uma thread em bg? Eu peguei um exemplo no Stack Overflow. Se eu rodar em background é possível que essa task de timeout seja "passada pra trás" se o device não tiver memória suficiente?